bin/put-multiarch.pl

```diff
@@ -209,6 +210,22 @@ ($targetRef, $sourceRef)
 				push @neededArtifactPromises, map { my $digest = $_->{digest}; sub { needed_artifacts_p($ref, $archRef->clone->digest($digest)) } } @$manifestListItems;
 			}
 
+			# sort Windows image manifests to ensure proper order in the image index
+			my $sorter = sub {
+				# also sort platform->variant for linux/arm?
+				for my $obj ($a, $b) {
+					return 0 unless $obj->{platform};
+					for my $field (qw{ os architecture os.version }) {
+						return 0 unless $obj->{platform}->{$field};
+					}
+				}
+				return 0 unless $a->{platform}->{os} eq 'windows' && $b->{platform}->{os} eq 'windows';
```

Comment:
```suggestion
				return 0 unless $a->{platform}->{os} eq $b->{platform}->{os};
```

Reply:
Ok, this seems fine. Any image index we are creating with two items having a matching `os` + `arch` that also has an `os.version` should be sorted by `os.version`.

Also swapped `}->{` to just `}{` since nowhere else in the file has cuddled arrows.